test/shared/build.gradle

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+apply plugin: 'es.hadoop.build.base'
+apply plugin: 'java'
+
+repositories {
+    mavenCentral()
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+dependencies {
+    implementation(project(':elasticsearch-hadoop-mr'))
+    
+    implementation(project.ext.hadoopClient)
```

Comment:
I suspect some of these probably _should_ be `api` dependencies as these test classes and fixtures expose Hadoop types in their APIs, [for example](https://github.com/jbaiera/elasticsearch-hadoop/blob/add-test-libs-project/test/shared/src/main/java/org/elasticsearch/hadoop/mr/PrintStreamOutputFormat.java#L38). This probably only compiles and works as-is incidentally because the consuming projects bring in hadoop already.

Reply:
That's a good point. I'll put that in for correctness.